ui/shared/components/panel/LoadWorkspaceDataForm.jsx

```diff
@@ -93,27 +93,48 @@ const DATA_BUCK_FIELD = {
 
 const REQUIRED_GENOME_FIELD = { ...GENOME_VERSION_FIELD, validate: validators.required }
 
-const postWorkspaceValues = (path, formatVals) => onSuccess => ({ namespace, name, ...values }) => (
-  new HttpRequestHelper(`/api/create_project_from_workspace/${namespace}/${name}/${path}`, onSuccess).post(
-    formatVals ? formatVals(values) : values,
-  ))
-
-const createProjectFromWorkspace = postWorkspaceValues(
-  'submit', ({ uploadedFile, ...values }) => ({ ...values, uploadedFileId: uploadedFile.uploadedFileId }),
-)((responseJson) => {
-  window.location.href = `/project/${responseJson.projectGuid}/project_page`
-})
-
-const FORM_WIZARD_PAGES = [
-  { fields: [AGREE_CHECKBOX], onPageSubmit: postWorkspaceValues('grant_access') },
-  { fields: [DATA_BUCK_FIELD, SAMPLE_TYPE_FIELD, REQUIRED_GENOME_FIELD], onPageSubmit: postWorkspaceValues('validate_vcf') },
+const postWorkspaceValues = (path, formatVals, formatUrl) => onSuccess => (
+  { workspaceNamespace, workspaceName, ...values },
+) => (
+  new HttpRequestHelper(
+    formatUrl ? formatUrl(values) : `/api/create_project_from_workspace/${workspaceNamespace}/${workspaceName}/${path}`,
+    onSuccess,
+  ).post(formatVals ? formatVals(values) : values)
+)
+
+const postSubmitValues = formatUrl => postWorkspaceValues(
+  'submit', ({ uploadedFile, ...values }) => ({ ...values, uploadedFileId: uploadedFile.uploadedFileId }), formatUrl,
+)
+
+const createProjectFromWorkspace = postSubmitValues()
+
+const addDataFromWorkspace = postSubmitValues(({ projectGuid }) => (`/api/project/${projectGuid}/add_workspace_data`))
+
+// eslint-disable-next-line no-return-assign
```

Comment:
we didn't need to supress linting before, so theres no reason we should have to now